devicon.json

```diff
@@ -1923,6 +1923,33 @@
         "color": "#092e20",
         "aliases": []
     },
+    {
+        "name": "djangorest",
+        "altnames": [
+            "Django REST framework"
+        ],
+        "tags": [
+            "framework",
+            "rest",
+            "api",
+            "python",
+            "web"
+        ],
+        "versions": {
+            "svg": [
+                "original",
+                "plain",
+                "original-wordmark",
+                "plain-wordmark"
+            ],
+            "font": [
+                "plain",
+                "plain-wordmark"
+            ]
+        },
+        "color": "#a30000",
+        "aliases": []
```

Comment:
Instead of duplicating the SVG's you can add them as aliases here. 
See this part in the wiki to learn more: https://github.com/devicons/devicon/wiki/Updating-%60devicon.json%60#aliases-and-aliasobj
```suggestion
        "aliases": [
            {
                "base": "original-wordmark",
                "alias": "original"
            },
            {
                "base": "plain-wordmark",
                "alias": "plain"
            }
        ]
```

Btw I think it should be possible to convert the plain icons into line icons if you want to add those too. Up to you if you wanna spend time on it or not though :)
I believe @kilian-paquier knows how to do it, if you need help

Reply:
I got your idea, but I have some proposals:

`original` - is "django" and "framework" texts, styled word "rest" centered and some sketch lines on the background.

`original-wordmark` - same as original, but without sketch lines on the background. But since they look quite the same I got this idea:

From `original` I can make a new `original-wordmark` with the word "rest" filled, looking more like a font.
![bitmap](https://user-images.githubusercontent.com/76992016/193537108-ab225f9f-cfec-47a3-bbee-18a15206c30c.png)

The same idea can be applied on `plain-wordmark`.

If it's ok I'll do this, otherwise I'll commit your changes and delete the similar `.svg` files.

Reply:
Hmm, I while I like the wordmark version you made, I don't think we should modify their logo to be honest. Imo we should just use the original and have an alias from original to original-wordmark.

I'd like to get some more opinions on this though.
@Panquesito7 @BenSouchet What do you guys think of @lunatic-fox 's proposal?

Reply:
Hi,
In my opinion the `original` is the one **with** the sketch lines, so in fine won't be in the generated devicon font due to no very lisible lines.
The `plain` should be an alias to the `plain-wordmark` and should be the logo **without** the sketch lines.
So in conclusion two different icons, the `original` and the `plain` / `plain-wordmark`.

Reply:
Seems alright to me!

Just to be extra sure:
`original` - Keep the existing one.
`plain` - The `plain` version of `original` without background sketch lines.
`plain-wordmark` - An alias of `plain` version.

Reply:
> Hi, In my opinion the `original` is the one **_with_** the sketch lines, so won't be in the generated devicon font due to no very lisible lines. The `plain` should be an alias to the `plain-wordmark` and should be the logo **_without_** the sketch lines. So in conclusion two different icons, the `original` and the `plain` / `plain-wordmark`.

Agreed with the `original` icon, it should be the one **with** the sketch lines.

I'm not so sure about the plain icon though. @lunatic-fox If I understand it correctly, the one **without** sketchlines is an icon you've made yourself right? That doesn't sit right with me, since most companies don't like it when you change their logos.

### In summary I would suggest the following:
- `original`:  Keep the one **with** sketchlines and colored text
- `original-wordmark`: An alias of `original`
- `plain`: Same as `original`, but mono-colored
- `plain-wordmark`: An alias of `plain`

Reply:
I've discarded my last proposal of filling the "rest" word and stick with this idea:
![bitmap1](https://user-images.githubusercontent.com/76992016/193681249-d5effd9c-9a73-40f5-b6cb-d04d691559d4.png)

Reply:
@Snailedlt I'm not as stuck on the idea of editing a bit logo to create icon for Devicon since, for Django REST it's an logo not an icon, so for me removing the sketchlines isn't "editing" the logo, someone looking at it still recognize Django REST logo and it's fit correctly with Devicon.
Agree that `original-wordmark` could be an alias to `original` but since it won't be added to the final generated font it's doesn't matter too much.
The last proposition of @lunatic-fox is for me the best in my option / compromise.

Reply:
@BenSouchet I think we just disagree then. I don't see any advantage of adding an icon which is not the official one, when we have the official one right there.

I think @Panquesito7 should also help decide though, so I'll wait for him to respond here

Reply:
A **distinction** need to be made between a logo (what we have as entry point with Django REST), and a icon which must be able to be used in small (_favicon for example_), even more if the icon will be added as glyph in a font, with icons the more unnecessary details are added the less you have in readability.

Also if Django REST aren't happy by the small edit made to add the logo as icon to Devicon they can comment / open an issue / contact.

Other point, the logo isn't registered / trademarked so this isn't any illegal action at all.

Lastly, you can contact them (Django REST community) via the official Github repo asking them if its okay or not to do at small simplification, so the readability of the glyph would be better in Devicon generated font. Or do they have a simplified version that they use for icon / favicon.

Reply:
Hello, is there still a need of information for the outline part ?

Reply:
> > Hi, In my opinion the `original` is the one **_with_** the sketch lines, so won't be in the generated devicon font due to no very lisible lines. The `plain` should be an alias to the `plain-wordmark` and should be the logo **_without_** the sketch lines. So in conclusion two different icons, the `original` and the `plain` / `plain-wordmark`.
> 
> @BenSouchet Thanks for the review 🙇 I agree with the `original` icon, it should be the one **with** the sketch lines.
> 
> I'm not so sure about the plain icon though. @lunatic-fox If I understand it correctly, the one **without** sketchlines is an icon you've made yourself right? That doesn't sit right with me, since most companies don't like it when you change their logos.
> 
> ### In summary I would suggest the following:
> * `original`:  Keep the one **with** sketchlines and colored text
> * `original-wordmark`: An alias of `original`
> * `plain`: Same as `original`, but mono-colored
> * `plain-wordmark`: An alias of `plain`

This sounds good to me. 👍

Reply:
@kilian-paquier 
> Hello, is there still a need of information for the outline part ?

Yeah, would love to hear your opinion too :)

Reply:
@kilianpaquier Please take a look when you have time :)

Reply:
> @kilianpaquier Please take a look when you have time :)

Hello, thank you for the reminder. You need both "django" and "framework" outlined ?

Reply:
@kilianpaquier
I'll leave that up to you to decide. As long as it adheres to our SVG-Standards and SVG-Versions wiki pages :)

Reply:
Here you go. Tell me if it's fine ?

![svgviewer-output-outline](https://user-images.githubusercontent.com/107955904/199091809-2c74d747-1fb5-47f8-b6d2-043a55c13ffa.svg)

Reply:
Love it!
@lunatic-fox what do you think? Should we add it?

Reply:
btw @kilianpaquier, what tools and methods do you use to make those clean line icons?

Reply:
I'll let you merge the appropriate paths and put the appropriate color

![svgviewer-output (1)](https://user-images.githubusercontent.com/107955904/199093742-60bb3866-ae0e-415e-b360-026ff63aa626.svg)

Reply:
I know the original does have the sketchy lines, but in that case, since it's original version I think that's understandable.

Reply:
@lunatic-fox I'm fine with that :)

Reply:
> I didn't keep the weird segments and dots from the logo. But it seems to be part of the official logo, I should add that version too.
> 
> @Snailedlt Inkscape. Maybe I should add a wiki part for the particularity of outline logos ?

Yes please, you should explain it in #1306 :D

Reply:
> @lunatic-fox I'm fine with that :)

Then, I think we have a consensus now.
So the PR would be:
`original` - with all the features
`plain` - without the sketchy lines

I think it's possible to discard `original-wordmark` and `plain-wordmark`, isn't it? And they can be aliased from `plain` version.

Reply:
hmmm, yesn't. What if I want the mono colored version of the original? :O I'm not sure. I think bottom line is that with only the versions we're working with in devicon we can't make everyone happy. 
So we either go with your first option:
1.  `wordmark` versions are without the sketchlines, and non-wordmark versions are with sketchlines.
2.  `original` has sketchlines, `plain` and `line` do not. Then we make aliases for `wordmark` versions.
3.  All of the non-wordmark versions have sketchlines. Then we make aliases for `wordmark` versions.

Personally leaning more towards 1. and 3. still. I kinda actually prefer 1. now that I think of it, since that leaves us with the most options, even if it's not entirely conventional

Reply:
I also think 1. makes almost as much conventional sense as 3., so definitely the best option for me

Reply:
Loved it! I'll make the change based on 1.

Just to confirm it'll be like:
`original` - regular icon
`plain` - monocolored original
`original-wordmark` - `original` without sketch lines
`plain-wordmark` - monocolored `original-wordmark`

Reply:
Great! Glad we can agree :D
I think it's good we had this discussion, definitely helps understand some more usecases, and intricacies of devicon

Reply:
> So we will be adding @kilianpaquier line versions?
Yes! 💯 
Make sure you add him as a co-author

Reply:
All done!
I was just refining some curves, because Inkscape sometimes do weird paths.

Reply:
Noice!
Btw, could you add an empty line at the end for each file? It's needed for some old IDE's

Reply:
> Noice! Btw, could you add an empty line at the end for each file? It's needed for some old IDE's

Hum, to me there's a empty line at the end of the file, maybe it's the Windows return carriage issue.
I'll be fixing that in a moment!

Reply:
I meant for the icons. Maybe, if you're using linux the return carriage is different than on windows.

Not a huge issue though, since there are tons of icons already without it. I kinda wanna add an empty line to all of them eventually though, so would be nice if future PR's had that empty line :)

Reply:
Got it! I'll be doing that later! 🙂

Reply:
Done via GitHub Codespaces!
It looks like GitHub cosmetically hides the final empty line. But if you try to edit the file directly, you will see this end line.

Reply:
Done via GitHub Codespaces!
It looks like GitHub cosmetically hides the final empty line. But if you try to edit the file directly, you will see this end line.

***\* Suggestion***
&emsp;Maybe we can add this code below in `.vscode/settings.json` of #1487, then it'll automatically insert an empty line at the end of a `xml` file **when saving it (trigger)**. `xml` language was selected, because `svg` is some kind of `xml` file and `svg` it's not specified as a language in `settings.json`.
> 
```json
"[xml]": {
  "files.insertFinalNewline": true,
}
```

Reply:
> It looks like GitHub cosmetically hides the final empty line

Yup, though it is possible to see still, since GitHub adds a little icon when the endline is missing:
![image](https://user-images.githubusercontent.com/43886029/199621519-01b46627-51d6-48c3-b49a-41025ed86735.png)

But it can be unreliable sometimes it seems. Notice the endline and no-endline icons in the same file here: 
![image](https://user-images.githubusercontent.com/43886029/199621543-b40997ba-15a8-4efd-8f91-ddafbc55ca3b.png)

Reply:
> _*** Suggestion**_  Maybe we can add this code below in `.vscode/settings.json` of #1487, then it'll automatically insert an empty line at the end of a `xml` file **when saving it (trigger)**. `xml` language was selected, because `svg` is some kind of `xml` file and `svg` it's not specified as a language in `settings.json`.
> 
> > 
> 
> ```json
> "[xml]": {
>   "files.insertFinalNewline": true,
> }
> ```

Sounds like a good idea, just contemplating whether we want endlines added to all files, or just xml files